app/common/util/src/main/java/io/syndesis/common/util/cache/LRUSoftCache.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.common.util.cache;
+
+import java.lang.ref.Reference;
+import java.lang.ref.SoftReference;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A special cache using soft references to hold data, to allow the garbage collector to reclaim space if the application
+ * is suffering for high memory usage.
+ */
+public class LRUSoftCache<K, V> implements Cache<K, V> {
+
+    private Cache<K, Reference<V>> cache;
+
+    public LRUSoftCache(int maxElements) {
+        this.cache = new LRUDefaultCache<>(maxElements);
+    }
+
+    @Override
+    public V get(K key) {
+        return getOptional(key)
```

Comment:
maybe we can get rid of the optional thing here: the optional object being created by getOptional does not provide any real value